client/state/themes/actions.js

```diff
@@ -167,15 +168,47 @@ export function themeActivated( theme, site ) {
  * Returns an action object to be used in signalling that a theme activation
  * has failed
  *
- * @param  {Object}  theme Theme received
- * @param  {Object}  site Site used for activation
+ * @param  {Number}  themeId Theme received
+ * @param  {Number}  siteId Site used for activation
+ * @param  {Number}  error Error response from server
  * @return {Object}  Action object
  */
-export function themeActivationFailed( theme, site, error ) {
+export function themeActivationFailed( themeId, siteId, error ) {
 	return {
 		type: THEME_ACTIVATE_FAILURE,
-		theme,
-		site,
+		themeId,
+		siteId,
 		error,
 	};
 }
+
+const defaultTrackData = {
+	theme: undefined,
+	previous_theme: 0,
+	source: 'unknown',
+	purchased: false,
+	search_term: null
+};
+
+export function activateTheme( themeId, siteId, trackThemesActivationData = defaultTrackData ) {
```

Comment:
I don't think the default really makes a lot of sense here since we'll always be passing a param for tracking here. If it's just for testing, I think we should rather create a dummy object in tests.